homeassistant/components/tado/__init__.py

```diff
@@ -169,6 +170,29 @@ def __init__(self, hass, username, password, fallback):
             "zone": {},
         }
 
+    def decide_overlay_mode(
+        self, duration: str, zone_id: int, overlay_mode: str | None = None
+    ) -> str:
+        """Return correct overlay mode based on the action and defaults."""
+        # If user gave duration then overlay mode needs to be timer
```

Comment:
Thanks for your good contribution on this!

Why not go for early returns principle? If an if-statement succeeds, it can directly return it. It could save a few lines of code.

Reply:
You are right, for the first if-statement it can return early. But not for the other two. I will make the change